configure/jaeger/README.md

```diff
@@ -1,207 +1,57 @@
 # Jaeger
 
-[Jaeger](https://github.com/jaegertracing/jaeger) is an open-source request tracing system that can
-run inside of your Kubernetes cluster. Sourcegraph can connect to Jaeger to provide real-time
-request traces that are useful for diagnosing performance issues and prescribing the appropriate
-remedy (typically scaling up one of the services). If you are experiencing performance issues with
-Sourcegraph, anticipate a high volume of traffic, or have a large amount of code, we recommend
-connecting Sourcegraph to a Jaeger instance.
+Jaeger is installed if you followed the standard installation instructions. Its resource objects are
+defined in the `base/jaeger` directory. Additionally, a Jaeger agent sidecar container is present in
+the `*.Deployment.yaml` and `*.StatefulSet.yaml` of certain services.
 
-There are two options for connecting Sourcegraph to Jaeger:
+## Connect Sourcegraph to an external Jaeger instance
 
-* [Deploy-a-new-Jaeger-instance-inside-the-Sourcegraph-cluster](#Deploying-a-new-Jaeger-instance-alongside-Sourcegraph)
-* [Connect to an existing Jaeger instance](#Connecting-Sourcegraph-to-an-existing-Jaeger-instance)
+If you have an existing Jaeger instance you would like to connect Sourcegraph to (instead of running
+the Jaeger instance inside the Sourcegraph cluster), do:
 
-If you are unsure of what to do, we recommend deploying a new Jaeger instance inside the Sourcegraph
-cluster.
+1. Remove the `base/jaeger` directory: `rm -rf base/jaeger`
+1. Update the Jaeger agent containers to point to your Jaeger collector.
+   1. Find all instances of Jaeger agent (`grep -R 'jaegertracing/jaeger-agent'`).
+   1. Update the `args` field of the Jaeger agent container configuration to point to the external
+      collector. E.g.,
+      ```
+      args:
+        - --reporter.grpc.host-port=external-jaeger-collector-host:14250
+        - --reporter.type=grpc
+      ```
+1. Apply these changes to the cluster.
 
-## Deploying a new Jaeger instance alongside Sourcegraph
+## Disable Jaeger entirely
 
-1. [Install the Jaeger
-   Operator](https://www.jaegertracing.io/docs/1.16/operator/#installing-the-operator-on-kubernetes)
-   in the Kubernetes cluster.
+To disable Jaeger entirely, do:
 
-1. Deploy Jaeger as a Custom Resource Definition object:
+1. Update the Sourcegraph [site
+   configuration](https://docs.sourcegraph.com/admin/config/site_config) to remove the
+   `observability.tracing` field.
+1. Remove the `base/jaeger` directory: `rm -rf base/jaeger`
+1. Remove the jaeger agent containers from each `*.Deployment.yaml` and `*.StatefulSet.ayml` file.
+1. Apply these changes to the cluster.
```

Comment:
can we provide kustomizations for these? This seems quite labour intensive for our admins + asking for merge conflicts down the road.

Reply:
Kustomizations aren't composable, which means they aren't ideal for the use case of turning on and off different components in the Sourcegraph k8s distribution. In addition, `kubectl apply -k` has surprising behavior such as `--prune` no longer working, which is surprising for admins who currently don't use kustomize. For these reasons, we are actually rethinking our decision to rely more heavily on kustomize, so manual instructions with the possibility of merge conflicts seems like the best option for now.

(Aside: Helm might actually be close to what we want than kustomize, now that it no longer includes tiller.)